crates/bevy_ecs/src/system/query.rs

```diff
@@ -651,7 +895,7 @@ where
     /// this call does not result in multiple mutable references to the same component
     #[inline]
     pub unsafe fn get_unchecked(
-        &'s self,
+        &self,
```

Comment:
This is kind of unrelated fix in this pull, so maybe add one line in the "incidental" part of the pull description about it.

Reply:
The elided lifetime should be the same as `'s`, so I don't think it's necessary to call out separately.

That said, I'd like to keep the explicit lifetime around for now; reasoning about this area of the code base's lifetimes was pretty gnarly.